src/many-sources/abstract-pool.js

```diff
@@ -34,8 +35,15 @@ inherit(AbstractPool, Stream, {
       if (this._queueLim === -1 || this._queue.length < this._queueLim) {
         this._addToQueue(toObs(obj));
       } else if (this._drop === 'old') {
-        this._removeOldest();
-        this._add(obj, toObs);
+        if (this._overlapping) {
+          this._addToCur(toObs(obj));
+          while (this._curSources.length > this._concurLim) {
+            this._removeOldest();
+          }
```

Comment:
I'm not thats quite equivalent, I need to think about it more, and there may not be a public api to aggravate the condition, but I think that when `{ concurLim: 1, queueLim: 1, drop: 'old' }`, and given that both `_curSources` and `_queue` are full and I add another one, I would want to:

1. drop the eldest item out of `_curSources`
2. promote the eldest item from `_queue`
3. add_ the new_ stream to the `_queue`

But, I'll need to think about it more. However, I think the same concern is valid for the overlapping side of the branch. If there _is_ a queue, then one should add to `_queue` and let `_removeOldest` pull it out of the queue, but that doesn't really resolve the actual use-case for using overlapping to begin with. I think I need to step back and think about it harder.